src/jrd/tra.cpp

```diff
@@ -3579,6 +3585,15 @@ void jrd_tra::rollbackToSavepoint(thread_db* tdbb, SavNumber number)
  *
  **************************************/
 {
+#ifdef SAVEPOINT_DEBUG
+	printf("Request to rollback savepoint %" QUADFORMAT "d. Stack: ", number);
+	for (Savepoint* itr = tra_save_point; itr; itr = itr->getNext())
```

Comment:
Please, don't use printf for debug purposes. Use common way and define appropriate (for ex SAV_trace) routine